File: HealthSysHub.Web.Utility/Models/PharmacyOrderDetails.cs
namespace HealthSysHub.Web.Utility.Models
{
    public class PharmacyOrderDetails
    {
        public PharmacyOrderDetails()
        {
            pharmacyOrderItemDetails = new List<PharmacyOrderItemDetails>();
        }
        public Guid PharmacyOrderId { get; set; }
        public Guid? PharmacyOrderRequestId { get; set; }
        public Guid? PharmacyId { get; set; }
        public string? OrderReference { get; set; }
        public string? Name { get; set; }
        public string? Phone { get; set; }
        public string? DoctorName { get; set; }
        public long? ItemQty { get; set; }
        public decimal? TotalAmount { get; set; }
        public decimal? DiscountAmount { get; set; }
        public decimal? FinalAmount { get; set; }
        public decimal? BalanceAmount { get; set; }
        public string? Notes { get; set; }
        public string? Status { get; set; }
        public Guid? CreatedBy { get; set; }
        public DateTimeOffset? CreatedOn { get; set; }
        public Guid? ModifiedBy { get; set; }
        public DateTimeOffset? ModifiedOn { get; set; }
        public bool IsActive { get; set; }
        public List<PharmacyOrderItemDetails> pharmacyOrderItemDetails { get; set; }
    }
}

File: HealthSysHub.Web.Pharmacy.UI/src/app/models/pharmacy-order-details.ts
import { PharmacyOrderItemDetails } from "./pharmacy-order-item-details";

export interface PharmacyOrderDetails {
  pharmacyOrderId: string; // Guid as string
  pharmacyOrderRequestId?: string | null; // Guid as string
  pharmacyId?: string | null; // Guid as string
  orderReference?: string | null;
  name?: string | null;
  phone?: string | null;
  doctorName?: string | null;
  itemQty?: number | null; // long as number
  totalAmount?: number | null; // decimal as number
  discountAmount?: number | null; // decimal as number
  finalAmount?: number | null; // decimal as number
  balanceAmount?: number | null; // decimal as number
  notes?: string | null;
  status?: string | null;
  createdBy?: string | null; // Guid as string
  createdOn?: string | null; // DateTimeOffset as ISO string
  modifiedBy?: string | null; // Guid as string
  modifiedOn?: string | null; // DateTimeOffset as ISO string
  isActive: boolean;
  pharmacyOrderItemDetails: PharmacyOrderItemDetails[];
}